common-ktx/src/test/java/co/nimblehq/common/extensions/DateExtensionsTest.kt

```diff
@@ -0,0 +1,339 @@
+        )
+    }
+
+    @Test
+    fun beginningOfMonth() {
+        val date = Date().with(year = 1987, month = 6, day = 2, hour = 5, minute = 0, second = 0)
+        assertEquals(
+            Date().with(year = 1987, month = 6, day = 1, hour = 0, minute = 0, second = 0, millisecond = 0),
+            date.beginningOfMonth
+        )
+    }
+
+    @Test
+    fun endOfMonth() {
+        val date = Date().with(year = 1987, month = 6, day = 2, hour = 12, minute = 0, second = 0)
+        assertEquals(
+            Date().with(year = 1987, month = 6, day = 30, hour = 23, minute = 59, second = 59),
+            date.endOfMonth
+        )
+    }
+
+    @Test
+    fun beginningOfDay() {
+        val date = Date().with(year = 1987, month = 6, day = 2, hour = 12, minute = 0, second = 0)
+        assertEquals(
+            Date().with(year = 1987, month = 6, day = 2, hour = 0, minute = 0, second = 0, millisecond = 0),
+            date.beginningOfDay
+        )
+    }
+
+    @Test
+    fun endOfDay() {
+        val date = Date().with(year = 1987, month = 6, day = 2, hour = 9, minute = 0, second = 0)
+        assertEquals(
+            Date().with(year = 1987, month = 6, day = 2, hour = 23, minute = 59, second = 59, millisecond = 999),
+            date.endOfDay
+        )
+    }
+
+    @Test
+    fun beginningOfHour() {
+        val date = Date().with(year = 1987, month = 6, day = 2, hour = 12, minute = 30, second = 30)
+        assertEquals(
+            Date().with(year = 1987, month = 6, day = 2, hour = 12, minute = 0, second = 0, millisecond = 0),
+            date.beginningOfHour
+        )
+    }
+
+    @Test
+    fun endOfHour() {
+        val date = Date().with(year = 1987, month = 6, day = 2, hour = 12, minute = 30, second = 30)
+        assertEquals(
+            Date().with(year = 1987, month = 6, day = 2, hour = 12, minute = 59, second = 59, millisecond = 999),
+            date.endOfHour
+        )
+    }
+
+    @Test
+    fun beginningOfMinute() {
+        val date = Date().with(year = 1987, month = 6, day = 2, hour = 12, minute = 30, second = 30)
+        assertEquals(
+            Date().with(year = 1987, month = 6, day = 2, hour = 12, minute = 30, second = 0, millisecond = 0),
+            date.beginningOfMinute
+        )
+    }
+
+    @Test
+    fun endOfMinute() {
+        val date = Date().with(year = 1987, month = 6, day = 2, hour = 12, minute = 30, second = 30)
+        assertEquals(
+            Date().with(year = 1987, month = 6, day = 2, hour = 12, minute = 30, second = 59, millisecond = 999),
+            date.endOfMinute
+        )
+    }
+
+    @Test
+    fun to_String() {
+        val calendar = Calendar.getInstance()
+        calendar.set(Calendar.MINUTE, calendar.get(Calendar.MINUTE) + 5)
+        assertEquals(
+            SimpleDateFormat("yyyy-MM-dd HH").format(calendar.time),
+            5.minutes.forward.toString("yyyy-MM-dd HH")
+        )
+    }
+
+    @Test
+    fun isSunday() {
+        run {
```

Comment:
Would it be possible to get rid of `run`? 🤔  I am curious why do we need `run` block here.

Reply:
@hoangnguyen92dn I use `run` block to separate the case in a test function so that we can use the same variable inside it, just for easy reading  🙏